### PR TITLE
metaprogramming for boolean array values of 1

### PR DESCRIPTION
**Pull request by martinthomson**
I don't know if this is much better than repeating "0, " 255 times, but it was an interesting exercise.

### CODE REVIEW

**Comment by martinthomson** [comment]
See also https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=699491dd1c8558554987671e10a834fc

File: ipa-core/src/ff/boolean_array.rs [hunk 1]
```diff
@@ -307,73 +334,14 @@ store_impl!(U8, 64);
 store_impl!(U32, 256);
 
 //impl BA3
```
Comment (akoshelev) [inline ipa-core/src/ff/boolean_array.rs:336]:
```suggestion
```

or something more applicable to multiple implementations

**Review by akoshelev — APPROVED** [review]
this is cool. I hope I'll never need to change this code. Is there any impact on build time with this macro in place?

File: ipa-core/src/ff/boolean_array.rs [hunk 2]
```diff
@@ -55,7 +82,7 @@ macro_rules! boolean_array_impl {
 
     type Store = BitArr!(for $bits, in u8, Lsb0);
 
-            ///
+            /// A Bootean array with $bits bits.
```
Comment (benjaminsavage) [inline ipa-core/src/ff/boolean_array.rs:85]: Bootean is cute.

**Pull request merged**